src/components/form-elements/checkbox/Checkbox.stories.mdx

```diff
@@ -0,0 +1,335 @@
+                    extraBold
+                    transform="uppercase"
+                    type="span"
+                    color="text-gray-70"
+                    size="small"
+                  >
+                    {checkboxType.type}
+                  </Headline>
+                </td>
+                <td>
+                  <Flex alignItems="center" direction="column">
+                    <Checkbox {...checkboxType.props} errorMessage={null} />
+                    <Checkbox
+                      {...checkboxType.props}
+                      checked
+                      errorMessage={null}
+                    />
+                    <Checkbox
+                      {...checkboxType.props}
+                      indeterminate
+                      errorMessage={null}
+                    />
+                  </Flex>
+                </td>
+                <td>
+                  <Flex alignItems="center" direction="column">
+                    <Checkbox {...checkboxType.props}>Checkbox label</Checkbox>
+                    <Checkbox {...checkboxType.props} checked>
+                      Checkbox label
+                    </Checkbox>
+                    <Checkbox {...checkboxType.props} indeterminate>
+                      Checkbox label
+                    </Checkbox>
+                  </Flex>
+                </td>
+                <td>
+                  <Flex
+                    alignItems="center"
+                    style={{
+                      maxWidth: '300px',
+                    }}
+                    direction="column"
+                  >
+                    <Checkbox
+                      {...checkboxType.props}
+                      description="More detailed description about this element. You can
+                          use here even some formatting and links."
+                    >
+                      Checkbox label
+                    </Checkbox>
+                    <Checkbox
+                      {...checkboxType.props}
+                      checked
+                      description="More detailed description about this element. You can
+                          use here even some formatting and links."
+                    >
+                      Checkbox label
+                    </Checkbox>
+                    <Checkbox
+                      {...checkboxType.props}
+                      indeterminate
+                      description="More detailed description about this element. You can
+                          use here even some formatting and links."
+                    >
+                      Checkbox label
+                    </Checkbox>
+                  </Flex>
+                </td>
+              </tr>
+            ))}
+          </tbody>
+        </StoryVariantTable>
+      </div>
+    )}
+  </Story>
+</Canvas>
+
+### With custom label
+
+<Canvas>
+  <Story name="With custom label">
+    {args => (
+      <label htmlFor="myCheckboxID">
```

Comment:
There is no need to duplicate the same function implementation. `htmlFor`, `aria-labelledby` and nesting do exactly the same - link label to the input. So nesting is enough in that case